cmd/helm/upgrade.go

```diff
@@ -270,6 +275,25 @@ func (u *upgradeCmd) run() error {
 		helm.UpgradeWait(u.wait),
 		helm.UpgradeDescription(u.description))
 	if err != nil {
+		fmt.Fprintf(u.out, "UPGRADE FAILED\nROLLING BACK\nError: %v\n", prettyError(err))
+		if u.atomic {
+			rollback := &rollbackCmd{
+				out:          u.out,
+				client:       u.client,
+				name:         u.release,
+				dryRun:       u.dryRun,
+				recreate:     u.recreate,
+				force:        u.force,
+				timeout:      u.timeout,
+				wait:         u.wait,
+				description:  "",
+				revision:     releaseHistory.Releases[0].Version,
```

Comment:
Why are we relying on the latest release version here? The latest released version is not always the version that was last deployed and in a healthy state. We need to roll back to the last good state we were in prior to the upgrade.

Reply:
For those landing here, @bacongobbler is working to address this case in [this PR](https://github.com/helm/helm/pull/5395)